pyee/util.py

```diff
@@ -0,0 +1,147 @@
+    ``AsyncIOEventEmitter`` so that you may register async event handlers
+    in your ``asyncio`` app but still be able to receive events from the
+    underlying event emitter and call the underlying event emitter's existing
+    handlers. This trick will also often work for a deprecated
+    ``EventEmitter`` instance.
+
+    When called, ``uplift`` instantiates a new instance of ``cls``, passing
+    along any unrecognized arguments, and overwrites the ``emit`` method on
+    the ``underlying`` event emitter to also emit events on the new event
+    emitter and vice versa. In both cases, they return whether the ``emit``
+    method was handled by either emitter. Execution order prefers the event
+    emitter on which ``emit`` was called.
+
+    ``uplift`` also adds an ``unwrap`` method to both instances, either of
+    which will unwrap both ``emit`` methods when called.
+
+    The ``error_handling`` flag can be configured to control what happens to
+    unhandled errors:
+
+    - 'new': Error handling for the new event emitter is always used and the
+      underlying library's non-event-based error handling is inert.
+    - 'underlying': Error handling on the underlying event emitter is always
+      used and the new event emitter can not implement non-event-based error
+      handling.
+    - 'neither': Error handling for the new event emitter is used if the
+      handler was registered on the new event emitter, and vice versa.
+
+    Tuning this option can be useful depending on how the underlying event
+    emitter does error handling. The default is 'new'.
+
+   The ``proxy_new_listener`` option can be configured to control how
+    ``new_listener`` events are treated:
+
+    - 'forward': ``new_listener`` events are propagated from the underlying
+    - 'both': ``new_listener`` events are propagated as with other events.
+    - 'neither': ``new_listener`` events are only fired on their respective
+      event emitters.
+      event emitter to the new event emitter but not vice versa.
+    - 'backward': ``new_listener`` events are propagated from the new event
+      emitter to the underlying event emitter, but not vice versa.
+
+    Tuning this option can be useful depending on how the ``new_listener``
+    event is used by the underlying event emitter, if at all. The default is
+    'forward', since ``underlying`` may not know how to handle certain
+    handlers, such as asyncio coroutines.
+
+    Each event emitter tracks its own internal table of handlers.
+    ``remove_listener``, ``remove_all_listeners`` and ``listeners`` all
+    work independently. This means you will have to remember which event
+    emitter an event handler was added to!
+
+    Note that both the new event emitter returned by ``cls`` and the
+    underlying event emitter should inherit from ``BaseEventEmitter``, or at
+    least implement the interface for the undocumented ``_call_handlers`` and
```

Comment:
Note that this helper should only be applied to an underlying EE once and that the new ee should probably not be used as an underlying. In other words, uplifting twice will break.